main/ping.go

```diff
@@ -270,3 +298,207 @@ func pingWatcher() {
+   validFlags    uint16
+   squawk        uint16
+   altitude_type uint8
+   callsign      [9]byte
+   emitter_type  uint8
+   tslc          uint8
+}
+
+func mavLinkFormat(x []byte) {
+	msglenMavLink := x[1]
+	msgtypeMavLink := x[5]
+	if msgtypeMavLink == 246 && msglenMavLink >= 38 {
+		mavLink := MavlinkTrafficMessageFormat{}
+		pingUsbHeaderLen := 6
+		pingUsbHeaderCursor := 0
+		mavLink.ICAO_address = uint32(x[3+pingUsbHeaderLen+pingUsbHeaderCursor])<<24 | uint32(x[2+pingUsbHeaderLen+pingUsbHeaderCursor])<<16 | uint32(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint32(x[0+pingUsbHeaderLen+pingUsbHeaderCursor])
+		pingUsbHeaderCursor = 4
+		mavLink.lat = int32(uint32(x[3+pingUsbHeaderLen+pingUsbHeaderCursor])<<24 | uint32(x[2+pingUsbHeaderLen+pingUsbHeaderCursor])<<16 | uint32(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint32(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 8
+		mavLink.lon = int32(uint32(x[3+pingUsbHeaderLen+pingUsbHeaderCursor])<<24 | uint32(x[2+pingUsbHeaderLen+pingUsbHeaderCursor])<<16 | uint32(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint32(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 12
+		mavLink.altitude = int32(uint32(x[3+pingUsbHeaderLen+pingUsbHeaderCursor])<<24 | uint32(x[2+pingUsbHeaderLen+pingUsbHeaderCursor])<<16 | uint32(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint32(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 16
+		mavLink.heading = (uint16(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint16(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 18
+		mavLink.hor_velocity = (uint16(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint16(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 20
+		mavLink.ver_velocity = int16(uint16(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint16(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 22
+		mavLink.validFlags = (uint16(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint16(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 24
+		mavLink.squawk = (uint16(x[1+pingUsbHeaderLen+pingUsbHeaderCursor])<<8 | uint16(x[0+pingUsbHeaderLen+pingUsbHeaderCursor]))
+		pingUsbHeaderCursor = 26
+		mavLink.altitude_type = x[0+pingUsbHeaderLen+pingUsbHeaderCursor]
+		pingUsbHeaderCursor = 27
+
+		for j := 0; j < 9; j++ {
+			mavLink.callsign[j] = 0
+		}
+		for a, b := range x[0+pingUsbHeaderLen+pingUsbHeaderCursor:] {
+			if a > 8 || b == ' ' {
+				break
+			}
+			mavLink.callsign[a] = b
+		}
+		pingUsbHeaderCursor = 36
+		mavLink.emitter_type = x[0+pingUsbHeaderLen+pingUsbHeaderCursor]
+		pingUsbHeaderCursor = 37
+		mavLink.tslc = x[0+pingUsbHeaderLen+pingUsbHeaderCursor]
+		log.Printf("ICAO_address %06X lat: %d lon: %d alt: %d head: %d call: %s vspeed: %d speed: %d", mavLink.ICAO_address, mavLink.lat, mavLink.lon, mavLink.altitude, mavLink.heading, mavLink.callsign, mavLink.ver_velocity, mavLink.hor_velocity)
```

Comment:
Maybe put this in an `if globalSettings.DEBUG` block.. I try not to log lines per received messages, because it can quickly fill up the FS overlay.